src/range_witness.rs

```diff
@@ -37,6 +37,21 @@ impl RangeWitness {
             extension_degree: extension_degree.try_into()?,
         })
     }
+
+    /// Produce a byte representation of the witness data for use in transcript RNG seeding
+    /// Note that this does _not_ need to be canonical!
+    pub(crate) fn to_bytes(&self) -> Zeroizing<Vec<u8>> {
+        let mut bytes = Zeroizing::new(Vec::<u8>::new());
```

Comment:
nit(perf): could perform a single allocation (though it is a bit of a pain :smile: )

```suggestion
        let size = self.openings.iter().map(|o| size_of::<u64>() + o.r.len() * 32).sum();
        let mut bytes = Zeroizing::new(Vec::<u8>::with_capacity(size));
```